### ISSUE TITLE
Sentinel and Safe’s Identity.yaml need stricter matchers

### ISSUE THREAD

**Issue by v0lkan** [vmware-tanzu/secrets-manager#190]
What we have:

```yaml
apiVersion: spire.spiffe.io/v1alpha1
kind: ClusterSPIFFEID
metadata:
  name: vsecm-safe
spec:
  spiffeIDTemplate: "spiffe://vsecm.com\
    /workload/vsecm-safe\
    /ns/{{ .PodMeta.Namespace }}\
    /sa/{{ .PodSpec.ServiceAccountName }}\
    /n/{{ .PodMeta.Name }}"
  podSelector:
    matchLabels:
      app.kubernetes.io/name: vsecm-safe
      app.kubernetes.io/part-of: vsecm-system
  workloadSelectorTemplates:
  - "k8s:ns:{{ .PodMeta.Namespace }}"
  - "k8s:sa:{{ .PodSpec.ServiceAccountName }}"
```

what we need:

```yaml
apiVersion: spire.spiffe.io/v1alpha1
kind: ClusterSPIFFEID
metadata:
  name: vsecm-safe
spec:
  spiffeIDTemplate: "spiffe://vsecm.com\
    /workload/vsecm-safe\
    /ns/{{ .PodMeta.Namespace }}\
    /sa/{{ .PodSpec.ServiceAccountName }}\
    /n/{{ .PodMeta.Name }}"
  podSelector:
    matchLabels:
      app.kubernetes.io/name: vsecm-safe
      app.kubernetes.io/part-of: vsecm-system
  workloadSelectorTemplates:
  - "k8s:ns:vsecm-system"
  - "k8s:sa:vsecm-safe"
```

same for Sentinel too. the `workloadSelectorTemplates` section needs to change.

**Comment by v0lkan** [comment]
@abhishek44sharma This will require helm chart changes too.

Would you like to address it in a new helm chart version?

**Comment by abhishek44sharma** [comment]
> @abhishek44sharma This will require helm chart changes too.
> 
> Would you like to address it in a new helm chart version?

Thanks for this @v0lkan 🙇 
Yes, we don't want to overwrite released version.

**Comment by abhishek44sharma** [comment]
Fixed by #232 and #238
